### PR TITLE
TW-97: Create group chat

### PR DESCRIPTION
**Pull request by sherlockvn**
https://github.com/linagora/twake-on-matrix/assets/43041967/0fabcfc8-b7ab-4183-bce3-039536eb72aa

Simulator Screenshot - iPhone 14 Pro - 2023-05-27 at 00 06 26

### CODE REVIEW

File: lib/pages/chat/group_chat_empty_view.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          _ruleChanel(L10n.of(context)!.persistentChatHistory),
+          const SizedBox(height: 8),
+          _ruleChanel(L10n.of(context)!.multiInteractionThreadedDiscussion),
+          const SizedBox(height: 10),
+        ],
+      ),
+    );
+  }
+
+  Widget _ruleChanel(String rule) {
```
Comment (hoangdat) [inline lib/pages/chat/group_chat_empty_view.dart:65]:
```suggestion
  Widget _ruleChannel(String rule) {
```

File: lib/pages/contacts/presentation/model/presentation_contact.dart [hunk 2]
```diff
@@ -11,13 +11,16 @@ class PresentationContact extends Equatable {
 
   final ContactStatus? status;
 
+  final bool isOwner;
```
Comment (hoangdat) [inline lib/pages/contacts/presentation/model/presentation_contact.dart:14]: Should be a get function with active client Id?

File: lib/pages/new_group/new_group.dart [hunk 3]
```diff
@@ -69,15 +86,77 @@ class NewGroupController extends State<NewGroup> {
...
+      enableEncryption: isEnableEEEncryptionNotifier.value,
+      preset: isGroupPrivate 
+        ? CreateRoomPreset.privateChat
+        : CreateRoomPreset.publicChat,
+    );
+
+    VRouter.of(context).toSegments(['rooms', roomId]);
+  }
+
+  void onGroupPrivacyChanged(bool val) {
```
Comment (hoangdat) [inline lib/pages/new_group/new_group.dart:143]: Don't use a magic bool. Use named parameter

File: lib/pages/contacts/presentation/model/presentation_contact.dart (hunk 2 above)
Comment (sherlockvn) [reply lib/pages/contacts/presentation/model/presentation_contact.dart:14]: didn't get your point

Comment (hoangdat) [reply lib/pages/contacts/presentation/model/presentation_contact.dart:14]: this function maybe an extension, not all the case we need it :)

File: lib/pages/new_group/new_group.dart [hunk 4]
```diff
@@ -69,15 +100,30 @@ class NewGroupController extends State<NewGroup> {
...
-    fetchContactsController.dispose();
+  Set<PresentationContact> getAllContactsGroupChat() {
+    final newContactsList = {
+          PresentationContact(
+            displayName: "You",
+            matrixId: Matrix.of(context).client.userID,
+            isOwner: true,
+          )
+        };
+    newContactsList.addAll(getContactsListWithoutOwner(contactsList));
```
Comment (hoangdat) [inline lib/pages/new_group/new_group.dart:111]: `getContactsListWithoutOwner` should be `getSelectedContacts`

File: lib/pages/new_group/new_group.dart [hunk 5]
```diff
@@ -53,6 +73,17 @@ class NewGroupController extends State<NewGroup> {
     });
   }
 
+  void onSearchKeywordChanged() {
+    searchContactsController.onSearchKeywordChanged = (String text) {
+      if (text.isEmpty) {
+        fetchContactsController.fetchCurrentTomContacts();
+      }
+    };
```
Comment (hoangdat) [inline lib/pages/new_group/new_group.dart:81]: missing debounce

File: lib/utils/string_extension.dart [hunk 6]
```diff
@@ -47,4 +48,11 @@ extension StringCasingExtension on String {
   String toTomMatrixId() {
     return '@$this:tom-dev.xyz';
   }
+  
+  bool isOwner(BuildContext context) {
+    if (isEmpty) {
+      return false;
+    }
+    return Matrix.of(context).client.userID == this;
+  }
```
Comment (hoangdat) [inline lib/utils/string_extension.dart:57]: IMO, it should be belong to extensions of PresentationContact

**Review by hoangdat — APPROVED** [review]

**Pull request merged**